src/app/pages/home/home.component.html

```diff
@@ -37,7 +39,9 @@ <h2 class="text-lg mb-2">Popular items</h2>
     >
       <app-secondary-card class="mb-4" *ngFor="let foodItem of foodItems"
         [data]="foodItem"
-        (favouriteToggled)="toggleFavourite($event.id, $event.type)">
+        (favouriteToggled)="toggleFavourite($event.id, $event.type)"
+        (goToDetails)="goToDetails($event.id, $event.type)"
```

Comment:
1. Bad naming of method. Better would be `navigateToDetails`
2. You don't need this output, just use simple:
```suggestion
        (click)="navigateToDetails($event.id, $event.type)"
```

Reply:
3. You can even remove this `click` event, you can use 
```html
[routerLink]="[ '/food-items',  foodItem.id]"
```

Same for restaurants